lib/backend/dynamo/shards.go

```diff
@@ -38,13 +39,41 @@ type shardEvent struct {
 }
 
 func (b *DynamoDBBackend) asyncPollStreams(ctx context.Context) error {
-	err := b.pollStreams(ctx)
+	retry, err := utils.NewLinear(utils.LinearConfig{
+		Step: b.RetryPeriod / 10,
```

Comment:
There's a chance that "step" will be 0 here (given low enough retry period). Not sure how this linear backoff will behave in this case - could potentially retry indefinitely?

Reply:
I think it's possible if the retry period is less than 9ns what is not practical